drivers/census_tracking/app/models/census_tracking/worksheet.rb

```diff
@@ -0,0 +1,92 @@
+###
+# Copyright 2016 - 2021 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+module CensusTracking
+  class Worksheet
+    include ::Filter::FilterScopes
+    include ArelHelper
+
+    def initialize(filter)
+      @filter = filter
+    end
+
+    def projects
+      projects = GrdaWarehouse::Hud::Project.
+        preload(:organization).
+        viewable_by(@filter.user).
+        where(id: @filter.effective_project_ids)
+
+      projects.map { |project| [HUD.project_type(project.computed_project_type), project.organization.name, project.safe_project_name, project.id] }.
+        sort_by { |project| [project[0], project[1], project[2]] }
+    end
+
+    def clients_by_project(project_id, population_query)
+      service_scope.
+        in_project(project_id).
+        where(population_query)
+    end
+
+    def clients_by_project_type(project_type, population_query)
+      service_scope.
+        in_project_type(HUD.project_type(project_type, true)).
+        where(population_query)
+    end
+
+    def clients_by_population(population_query)
+      service_scope.where(population_query)
+    end
+
+    def populations  # rubocop:disable Metrics/AbcSize
+      @populations ||= {
```

Comment:
This doesn't perform when we get over ~10 projects or so, let's fetch in batches by project and then by project id and do the population calculations in ruby.